onebusaway-android/src/main/java/com/joulespersecond/oba/ObaAnalytics.java

```diff
@@ -0,0 +1,260 @@
+                        .setAction("Stop Id: " + stopId)
+                        .setLabel(stopDistance.toString())
+                        .setValue(1)
+                        .setCustomDimension(1, obaRegionName)
+                        .build());
+
+            }
+
+        } catch (Exception e) {
+            //If location comes null
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * For reporting activities on Start
+     *
+     * @param activity
+     */
+    public static void reportActivityStart(Activity activity) {
+        Tracker tracker = Application.get().getTracker(Application.TrackerName.APP_TRACKER);
+        tracker.setScreenName(activity.getClass().getSimpleName());
+        tracker.send(new HitBuilders.ScreenViewBuilder().build());
+
+        Tracker tracker2 = Application.get().getTracker(Application.TrackerName.GLOBAL_TRACKER);
+        tracker2.setScreenName(activity.getClass().getSimpleName());
+        tracker2.send(new HitBuilders.ScreenViewBuilder().build());
+    }
+
+
+    /**
+     * For reporting fragments on Start
+     *
+     * @param fragment
+     */
+    public static void reportFragmentStart(Fragment fragment) {
+        Tracker tracker = Application.get().getTracker(Application.TrackerName.APP_TRACKER);
+        tracker.setScreenName(fragment.getClass().getSimpleName());
+        tracker.send(new HitBuilders.ScreenViewBuilder().build());
+
+        Tracker tracker2 = Application.get().getTracker(Application.TrackerName.GLOBAL_TRACKER);
+        tracker2.setScreenName(fragment.getClass().getSimpleName());
+        tracker2.send(new HitBuilders.ScreenViewBuilder().build());
+    }
+
+    public static void initAnalytics(Context context) {
+        if (isAnalyticsActive() == false) {
+            //Disable Google Analytics
+            GoogleAnalytics.getInstance(context).setAppOptOut(true);
+        }
+
+        if (BuildConfig.DEBUG) {
+            //Disables reporting when app runs on debug
+            GoogleAnalytics.getInstance(context).setDryRun(true);
+        }
+    }
+
+    /**
+     * @return is GA enabled or disabled from settings
+     */
+    private static Boolean isAnalyticsActive() {
+        SharedPreferences settings = Application.getPrefs();
+        return settings.getBoolean(Application.get().getString(R.string.preferences_key_analytics), Boolean.TRUE);
+    }
+
+    private static String getObaRegionName() {
+        String regionName = null;
+        ObaRegion region = Application.get().getCurrentRegion();
+
+        if (region != null && region.getName() != null) {
+            regionName = region.getName();
+        } else if (Application.get().getCustomApiUrl() != null) {
+            MessageDigest digest = null;
+            try {
+                digest = MessageDigest.getInstance("SHA-1");
+                digest.update(Application.get().getCustomApiUrl().getBytes());
+                regionName = Application.getHex(digest.digest());
```

Comment:
I think the hashed custom API should be prefixed here by the `R.string.analytics_label_custom_url`?  This would allow us to return a list (or count) of all custom APIs that we've seen.  So, it would look like `CUSTOM_URL=8nh2j02f...`.  Otherwise, if we have only the SHA-1 hash, we'd need to filter for all custom API URLs that aren't region names, which would be much harder to implement/maintain.